proxy_agent_extension/src/handler_main.rs

```diff
@@ -252,6 +252,10 @@ fn enable_handler(status_folder: PathBuf, config_seq_no: &Option<String>) {
         }
         let mut count = 0;
         loop {
+            if process_running {
```

Comment:
Please move these lines outside the loop since you are not refreshing the process_running variable inside the loop, put them inside the loop is unnecessary. Put them around line 252.